core-services/image-mirroring/README.md

```diff
@@ -28,18 +51,29 @@ are currently targeted by existing mirroring jobs:
 Submit a PR adding a new subdirectory here, with at least a single mapping file
 and an `OWNERS` file (so that you can maintain your mappings). The mapping files
 should follow the `mapping_$name$anything` naming convention to avoid conflicts
-when put to a ConfigMap.
+when put to a ConfigMap. Submit a PR which contains the new mappings, and after merging,
+we can expect to see them when
+
+> oc get cm -n ci image-mirror-mappings -o yaml | yq -r '.data | keys[]'
+
+The rehearsal of the periodic job defined below requires the ConfigMap `image-mirror-mappings` to have the updated mappings.
 
 Additionally, you will need to add a new Periodic job
 [here](../../ci-operator/jobs/infra-image-mirroring.yaml).  You should not need
 to modify anything in the job besides the items marked as `FIXME`, where you
 just need to fill in the name of your image set (it should be the same as the
-name of the subdirectory here). To push images to an external repository, the
-job needs credentials. These credentials need to be available in the cluster as
+name of the subdirectory here).
+
+To push images to an external repository, the
+job needs credentials.
+Use `docker`/`podman` to generate `${HOME}/.docker/config.json` and send it with gpg-encryption it to DPTP. These credentials need to be available in the cluster as
 a secret. The secrets are stored in DPTP Bitwarden vault and then synced to the
 cluster: [talk to DPTP on Slack](https://coreos.slack.com/messages/CBN38N3MW)
 about storing the credentials in Bitwarden.
 
+The rehearsal will be triggered for the PR with the job above if it has label `pj-rehearse.openshift.io/can-be-rehearsed: "true"`.
```

Comment:
Encouraging people to rehearse the image mirroring job sounds a bit risky to me, may result in pushing who-knows-which images to who-know-where, especially with people copy-pasting code around... Are we sure we want to do that?

Reply:
Also this will fail if anyone attempts to add both mappings and jobs in the same PR. They would need to merge the mappings first (to get them to the CM in the cluster) and only then the rehearsal would work.

Personallly, I think discussing rehearsals is just confusing and does not bring enough value because of the complexities, I would just leave it out.

Reply:
Agreed

Reply:
I will remove the rehearsal part.

Just a comment on the rehearsal:
1. @davidvossel and I found issues in the job PR with the rehearsal tool. Could that be the use case of rehearsal?
2. skipping the rehearsal would not avoid the error caused by copy-pasting error. It will just postpone when we hit it (after the PR is merged, anything in rehearsal will happen in periodic).
3. _The rehearsal of the periodic job defined below requires the ConfigMap `image-mirror-mappings` to have the updated mappings._ above clarifies it is required for the rehearsal. We know it from the failing rehearsals too.